pyphare/pyphare/pharesee/hierarchy.py

```diff
@@ -539,12 +575,103 @@ def __init__(self, patch_levels, domain_box, refinement_ratio=2, time=0., data_f
         if data_files is not None:
             self.data_files.update(data_files)
 
+        if len(self.quantities()) > 1:
+            for qty in self.quantities():
+                if qty in self.__dict__:
+                    continue
+                first = True
+                for time, levels in self.time_hier.items():
+                    new_lvls = {}
+                    for ilvl, level in levels.items():
+                        patches = []
+                        for patch in level.patches:
+                            patches += [Patch({qty: patch.patch_datas[qty]})]
+                        new_lvls[ilvl] = PatchLevel(ilvl, patches)
+                    if first:
+                        self.__dict__[qty] = PatchHierarchy(
```

Comment:
sure. As I said the other day, this `.x` etc. is a VectorField property. A PatchHierarchy is more general, it could very well contain patch datas of diverse quantities that are unrelated with each other, and I have no other choice than using the patchdata name as an attribute.

We said the other day that in you VectorField class, you will have to rename these attributes yourself, knowing that you manipulate a 3 components vector, you can do `.x`, `.y`and `.z`, but I don't.